### PR TITLE
ast, expression: add builtin function: CHARACTER_LENGTH

### PR DESCRIPTION
**Pull request by spongedu**
According to https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_char-length, CHARACTER_LENGTH() is a synonym for CHAR_LENGTH().

currently, builtin function 'CHARACTER_LENGTH' dose not work. As follows:
tidb> select CHARACTER_LENGTH('a');
ERROR 1305 (42000): FUNCTION character_length does not exist

with this pr:
tidb> select CHARACTER_LENGTH('a');
+-----------------------+
| CHARACTER_LENGTH('a') |
+-----------------------+
|                     1 |
+-----------------------+
1 row in set (0.01 sec)

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -527,6 +527,17 @@ func (s *testIntegrationSuite) TestStringBuiltin(c *C) {
 	result = tk.MustQuery(`select bin("中文");`)
 	result.Check(testkit.Rows("0"))
 
+	// for character_length
+	result = tk.MustQuery(`select character_length(null);`)
```
Comment (XuHuaiyu) [inline expression/integration_test.go]: result = tk.MustQuery(`select character_length(null), character_length("Hello"), .....;`)
result.Check(testkit.Rows("<nil> 5 ...."))

File: expression/builtin.go [hunk 2]
```diff
@@ -662,53 +662,54 @@ var funcs = map[string]functionClass{
...
+	ast.Upper:           &upperFunctionClass{baseFunctionClass{ast.Upper, 1, 1}},
+	ast.Ucase:           &upperFunctionClass{baseFunctionClass{ast.Ucase, 1, 1}},
+	ast.Hex:             &hexFunctionClass{baseFunctionClass{ast.Hex, 1, 1}},
+	ast.Unhex:           &unhexFunctionClass{baseFunctionClass{ast.Unhex, 1, 1}},
+	ast.Rpad:            &rpadFunctionClass{baseFunctionClass{ast.Rpad, 3, 3}},
+	ast.BitLength:       &bitLengthFunctionClass{baseFunctionClass{ast.BitLength, 1, 1}},
+	ast.CharFunc:        &charFunctionClass{baseFunctionClass{ast.CharFunc, 2, -1}},
+	ast.CharLength:      &charLengthFunctionClass{baseFunctionClass{ast.CharLength, 1, 1}},
+	ast.FindInSet:       &findInSetFunctionClass{baseFunctionClass{ast.FindInSet, 2, 2}},
+	ast.CharacterLength: &charLengthFunctionClass{baseFunctionClass{ast.CharacterLength, 1, 1}},
```
Comment (zz-jason) [inline expression/builtin.go]: put it together with `ast.CharLength`

File: expression/integration_test.go (hunk 1 above)
Comment (spongedu) [inline expression/integration_test.go]: ok

File: expression/builtin.go (hunk 2 above)
Comment (spongedu) [inline expression/builtin.go]: ok

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by spongedu** [comment]
@zz-jason PTAL

**Comment by zz-jason** [comment]
to https://github.com/pingcap/tidb/issues/4080

**Pull request merged**